shopinvader_api_cart/routers/cart.py

```diff
@@ -236,5 +236,8 @@ def _update(self, partner, data, uuid):
         cart = self.env["sale.order"]._find_open_cart(partner.id, uuid)
         if not cart:
             cart = self.env["sale.order"]._create_empty_cart(partner.id)
-        cart.write(data.convert_to_sale_write())
+        cart.write(self._update_prepare_data(data))
         return cart
+
+    def _update_prepare_data(self, data):
+        return data.convert_to_sale_write()
```

Comment:
IMO we should modify the `convert_to_sale_write` method to pass the environment and avoid to scattering the logic to convert the model into a dict at different place.

Reply:
I agree... can I introduce a breaking change then? :grin:

Reply:
Yes

Reply:
I split this change to https://github.com/shopinvader/odoo-shopinvader/pull/1534
@lmignon could we fast track that?